src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class PurchaseHandler implements OperationHandler {
+    private static final String PURCHASE = "p";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(PURCHASE)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
+                }
+            }
+        }
+        return listOfFruits;
+    }
```

Comment:
The logic within the `PurchaseHandler` class seems to be incorrect. The purpose of a purchase operation is to reduce the quantity of fruit in storage, not to add it. The current implementation is adding quantities together, which is not the expected behavior for a purchase. Additionally, there is no check to ensure that the resulting balance is positive, which is a requirement according to the checklist.